src/abstract-interpretation/data-frame/mappers/function-mapper.ts

```diff
@@ -0,0 +1,1440 @@
+import type { FlowrConfigOptions } from '../../../config';
+import { defaultConfigOptions, VariableResolve } from '../../../config';
+import { type ResolveInfo } from '../../../dataflow/eval/resolve/alias-tracking';
+import type { DataflowGraph } from '../../../dataflow/graph/graph';
+import { toUnnamedArgument } from '../../../dataflow/internal/process/functions/call/argument/make-argument';
+import { findSource, getSourceProvider } from '../../../dataflow/internal/process/functions/call/built-in/built-in-source';
+import type { RNode } from '../../../r-bridge/lang-4.x/ast/model/model';
+import type { RFunctionArgument } from '../../../r-bridge/lang-4.x/ast/model/nodes/r-function-call';
+import { EmptyArgument } from '../../../r-bridge/lang-4.x/ast/model/nodes/r-function-call';
+import type { ParentInformation } from '../../../r-bridge/lang-4.x/ast/model/processing/decorate';
+import { RType } from '../../../r-bridge/lang-4.x/ast/model/type';
+import type { RParseRequest } from '../../../r-bridge/retriever';
+import { requestFromInput } from '../../../r-bridge/retriever';
+import { assertUnreachable, isNotUndefined, isUndefined } from '../../../util/assert';
+import { readLineByLineSync } from '../../../util/files';
+import type { DataFrameExpressionInfo, DataFrameOperation } from '../absint-info';
+import { DataFrameTop } from '../domain';
+import { resolveIdToArgName, resolveIdToArgValue, resolveIdToArgValueSymbolName, resolveIdToArgVectorLength, unescapeSpecialChars } from '../resolve-args';
+import type { ConstraintType } from '../semantics';
+import { resolveIdToDataFrameShape } from '../shape-inference';
+import type { FunctionParameterLocation } from './arguments';
+import { escapeRegExp, filterValidNames, getArgumentValue, getEffectiveArgs, getFunctionArgument, getFunctionArguments, getUnresolvedSymbolsInExpression, hasCriticalArgument, isDataFrameArgument, isNamedArgument, isRNull } from './arguments';
+
+/**
+ * Represents the different types of data frames in R
+ */
+enum DataFrameType {
+	DataFrame = 'data.frame',
+	Tibble = 'tibble',
+	DataTable = 'data.table'
+}
+
+/**
+ * Mapper for mapping the supported concrete data frame functions to mapper functions,
+ * including information about the origin library of the functions and the type of the returned data frame.
+ */
+const DataFrameFunctionMapper = {
+	'data.frame':    { mapper: mapDataFrameCreate },
```

Comment:
maybe make the return type explicit for all of them and add the library for a base version too?